common/os.cpp

```diff
@@ -261,12 +261,28 @@ Some auxillary functionalities might be affected. Please report this message if
         else
         {
             GUID folder;
+            HRESULT hr;
             switch (f)
             {
             case user_desktop: folder = FOLDERID_Desktop;
                 break;
             case user_documents: folder = FOLDERID_Documents;
-                break;
+                // Use for documents with SHGetFolderPath instead of SHGetKnownFolderPath because SHGetKnownFolderPath 
+                // Doesn't give the path to intel one drive documents folder but to the local documents folder
+                TCHAR path[MAX_PATH];
+                hr = SHGetFolderPath(NULL, CSIDL_PERSONAL, NULL, 0, path);
```

Comment:
Why not use the A version directly so we don't have to resort to weird conversions using `wstring`?